dev-qt/qtwebengine/qtwebengine-6.9999.ebuild

```diff
@@ -0,0 +1,271 @@
+"
+BDEPEND="
+	$(python_gen_any_dep 'dev-python/html5lib[${PYTHON_USEDEP}]')
+	dev-util/gperf
+	dev-util/ninja
+	dev-util/re2c
+	net-libs/nodejs[ssl]
+	sys-devel/bison
+	sys-devel/flex
+"
+
+python_check_deps() {
+	python_has_version "dev-python/html5lib[${PYTHON_USEDEP}]"
+}
+
+qtwebengine_check-reqs() {
+	[[ ${MERGE_TYPE} == binary ]] && return
+
+	# bug #307861
+	eshopts_push -s extglob
+	if is-flagq '-g?(gdb)?([1-9])'; then
+		ewarn "You have enabled debug info (probably have -g or -ggdb in your CFLAGS/CXXFLAGS)."
+		ewarn "You may experience really long compilation times and/or increased memory usage."
+		ewarn "If compilation fails, please try removing -g/-ggdb before reporting a bug."
+	fi
+	eshopts_pop
+
+	# (check-reqs added for bug #570534)
+	#
+	# Estimate the amount of RAM required
+	# Multiplier is *10 because Bash doesn't do floating point maths.
+	# Let's crudely assume ~2GB per compiler job for GCC.
+	local multiplier=20
+
+	# And call it ~1.5GB for Clang.
+	if tc-is-clang ; then
+		multiplier=15
+	fi
+
+	local CHECKREQS_DISK_BUILD="7G"
+	local CHECKREQS_DISK_USR="150M"
+	if ! has "distcc" ${FEATURES} ; then
+		# bug #830661
+		# Not super realistic to come up with good estimates for distcc right now
+		local CHECKREQS_MEMORY=$(($(makeopts_jobs)*multiplier/10))G
+	fi
+
+	check-reqs_${EBUILD_PHASE_FUNC}
+}
+
+pkg_pretend() {
+	qtwebengine_check-reqs
+}
+
+pkg_setup() {
+	qtwebengine_check-reqs
+	python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	qt6-build_src_prepare
+
+	# bug 620444 - ensure local headers are used
+	find . -type f -name "*.pr[fio]" -exec \
+		sed -i -e 's|INCLUDEPATH += |&$${QTWEBENGINE_ROOT}_build/include $${QTWEBENGINE_ROOT}/include |' {} + || die
+
+	if use system-icu; then
+		# Sanity check to ensure that bundled copy of ICU is not used.
+		# Whole src/3rdparty/chromium/third_party/icu directory cannot be deleted because
+		# src/3rdparty/chromium/third_party/icu/BUILD.gn is used by build system.
+		# If usage of headers of bundled copy of ICU occurs, then lists of shim headers in
+		# shim_headers("icui18n_shim") and shim_headers("icuuc_shim") in
+		# src/3rdparty/chromium/third_party/icu/BUILD.gn should be updated.
+		local file
+		while read file; do
+			echo "#error This file should not be used!" > "${file}" || die
+		done < <(
+			find src/3rdparty/chromium/third_party/icu -type f \
+				\( -name '*.c' -o -name '*.cpp' -o -name '*.h' \) 2>/dev/null
+		)
+	fi
+
+	# for www-plugins/chrome-binary-plugins (widevine) search paths on prefix
+	hprefixify -w /Gentoo/ src/core/content_client_qt.cpp
+
+	# store chromium versions, only used in postinst for a warning
+	local chromium
+	mapfile -t chromium < CHROMIUM_VERSION || die
+	[[ ${chromium[1]} =~ ^Based.*:[^0-9]+([0-9.]+$) ]] &&
+		QT6_CHROMIUM_VER=${BASH_REMATCH[1]} || die
+	[[ ${chromium[2]} =~ ^Patched.+:[^0-9]+([0-9.]+$) ]] &&
+		QT6_CHROMIUM_PATCHES_VER=${BASH_REMATCH[1]} || die
+}
+
+src_configure() {
+	export NINJA_PATH=${BROOT}/usr/bin/ninja
```

Comment:
Wonder if should allow ${NINJA} override w/ type -P or something but I'm not sure if we really care about testing it with samu.

Reply:
In my TODO I have a line about checking if I can just remove that altogether. I think this may be a remnant from the Qt5 ebuild (pre-cmake). Like qttools I didn't really look at qtwebengine yet.

Reply:
Yeah, from a quick look NINJA_PATH is ignored:
```
find_package(Ninja 1.7.2 REQUIRED)
<snip>
    COMMAND Ninja::ninja -C ${GN_BINARY_DIR}/$<CONFIG> ${GN_EXECUTABLE}
```
Looking at FindNinja.cmake, if we wanted alternatives I could pass -DNinja_EXECUTABLE="something" though -- but it also does `ninja --version` and may not be happy with samu as-is.

As noted in one of the commit message, do wish could just use dev-util/gn but qtwebengine uses a modified gn (unsure how compatible it is) and also insists on taking it from qt6 directories if using system's.

Reply:
After finding it out that NINJAFLAGS is ignored too (it uses it for a ninja command on windows, but not to build gn), I thought might as well try samu for :rocket:`SAMUFLAGS`:rocket: but what I feared did happen:
```
  Could NOT find Ninja: (Required is at least version "1.7.2") (found samu)
```

Reply:
Or wait it works now, wonder how that happened before :thinking: Well, there's still nested cmake calls that will require passing it differently though.
```
-- Found Ninja: /usr/bin/samu (found suitable version "1.9", minimum required is "1.7.2") 
```